aten/src/ATen/CPUGenerator.cpp

```diff
@@ -51,7 +51,9 @@ CPUGenerator::CPUGenerator(uint64_t seed_in)
   : Generator{Device(DeviceType::CPU)},
     engine_{seed_in},
     next_float_normal_sample_{c10::optional<float>()},
-    next_double_normal_sample_{c10::optional<double>()} { }
+    next_double_normal_sample_{c10::optional<double>()} {
+  key_set_ = DispatchKeySet(c10::DispatchKey::CPUTensorId);
```

Comment:
C++ nit: there is no reason to defer the initialization of `key_set_` into the body of the constructor, just directly initialize it as all the other fields are above